File: Graph/go/undirected_path.go
package graph

// UndirectedPath reports whether there is a path between src and dst in an undirected graph.
func UndirectedPath(edges [][2]string, src, dst string) bool {
	g := BuildUndirectedGraph(edges)
	visited := map[string]struct{}{}
	return undirectedDFS(g, src, dst, visited)
}

func undirectedDFS(g Graph, node, dst string, visited map[string]struct{}) bool {
	if node == dst {
		return true
	}
	if _, seen := visited[node]; seen {
		return false
	}
	visited[node] = struct{}{}

	for _, neighbor := range g[node] {
		if undirectedDFS(g, neighbor, dst, visited) {
			return true
		}
	}
	return false
}

File: Graph/php/undirectedPath.php
<?php

declare(strict_types=1);

require_once __DIR__ . '/GraphUtils.php';

function undirectedPath(array $edges, string $src, string $dst): bool
{
    $graph = buildUndirectedGraph($edges);
    $visited = [];
    return undirectedPathDfs($graph, $src, $dst, $visited);
}

function undirectedPathDfs(array $graph, string $node, string $dst, array &$visited): bool
{
    if ($node === $dst) {
        return true;
    }
    if (isset($visited[$node])) {
        return false;
    }

    $visited[$node] = true;
    foreach ($graph[$node] ?? [] as $neighbor) {
        if (undirectedPathDfs($graph, $neighbor, $dst, $visited)) {
            return true;
        }
    }
    return false;
}
